walkeeper/src/send_wal.rs

```diff
@@ -293,12 +307,18 @@ impl ReplicationConn {
                 if let Some(lsn) = lsn {
                     end_pos = lsn;
                 } else {
-                    // Is is time to end streaming to this replica?
+                    // Is it time to end streaming to this replica?
                     if spg.timeline.get().check_stop_streaming(replica_id) {
-                        let timelineid = spg.timeline.get().timelineid;
+                        // TODO this expect is ugly, will it be better with node id?
+                        //   can we guarantee that this code is not executed during walproposer recovery?
+                        let pageserver_connstr = pageserver_connstr
```

Comment:
@lubennikovaav what do you think about this `expect` call? It was not triggered in tests so I assume this code path it not followed in walproposer recovery but can we guarantee somehow that recovery won't touch this?

Reply:
If walproposer is in recovery `stop_lsn` is set and this is checked in the code above this line.